packages/frontend/web-ui/src/auth/hooks/useAuthForgot.ts

```diff
@@ -0,0 +1,178 @@
+    });
+  };
+
+  function buildValidation(): AuthForgotFormValidationResult {
+    const formValidation: AuthForgotFormValidationResult = {};
+
+    const emailValidation: Either<string, undefined> = validateEmail(
+      authForgotData.form.fields.email,
+    );
+
+    if (!emailValidation.isRight) {
+      formValidation.email = emailValidation.value;
+    }
+
+    return formValidation;
+  }
+
+  function createUserCode(): void {
+    setAuthForgotData({
+      form: {
+        fields: {
+          ...authForgotData.form.fields,
+        },
+        validation: { ...authForgotData.form.validation },
+      },
+      status: UseAuthForgotStatus.sendRequest,
+    });
+
+    void triggerCreateUserCode({
+      params: [
+        {
+          email: authForgotData.form.fields.email,
+        },
+        {
+          kind: 'resetPassword',
+        },
+      ],
+    });
+  }
+
+  function onSubmit(event: React.FormEvent): void {
+    event.preventDefault();
+
+    if (authForgotData.status !== UseAuthForgotStatus.idle) {
+      return;
+    }
+
+    const formValidation: AuthForgotFormValidationResult = buildValidation();
+
+    if (Object.keys(formValidation).length === 0) {
+      setAuthForgotData({
+        form: {
+          fields: {
+            ...authForgotData.form.fields,
+          },
+          validation: formValidation,
+        },
+        status: authForgotData.status,
+      });
+
+      createUserCode();
+    } else {
+      setAuthForgotData({
+        form: {
+          fields: {
+            ...authForgotData.form.fields,
+          },
+          validation: formValidation,
+        },
+        status: UseAuthForgotStatus.idle,
+      });
+    }
+  }
+
+  useEffect(() => {
+    if (userCodeCreatedResult !== null) {
+      if (userCodeCreatedResult.isRight) {
+        setAuthForgotData({
+          form: {
+            fields: {
+              ...authForgotData.form.fields,
+            },
+            validation: { ...authForgotData.form.validation },
+          },
+          status: UseAuthForgotStatus.success,
+        });
+      } else {
+        if (isSerializableAppError(userCodeCreatedResult.value)) {
+          setAuthForgotData({
+            form: {
+              errorMessage: getCreateUserCodeErrorMessage(
+                userCodeCreatedResult.value.kind,
+              ),
+              fields: {
+                ...authForgotData.form.fields,
+              },
+              validation: { ...authForgotData.form.validation },
+            },
+            status: UseAuthForgotStatus.backendError,
+          });
+        } else {
+          setAuthForgotData({
+            form: {
+              errorMessage: getCreateUserCodeErrorMessage(undefined),
+              fields: {
+                ...authForgotData.form.fields,
+              },
+              validation: { ...authForgotData.form.validation },
+            },
+            status: UseAuthForgotStatus.backendError,
+          });
+        }
```

Comment:
In both cases, all the fields are the same except for `errorMessage`. Let's add a build error message function in this module so we don't need to duplicate that much code:

```suggestion
        const errorMessage: string = buildErrorMessage(userCodeCreatedResult.value);
        setAuthForgotData({
          form: {
            errorMessage,
            fields: {
              ...authForgotData.form.fields,
            },
            validation: { ...authForgotData.form.validation },
          },
          status: UseAuthForgotStatus.backendError,
        });
```

Reply:
Done!